Improve qvm-check mechanism on invalid domains

alimirjamali opened the pull request:
fixes: https://github.com/QubesOS/qubes-issues/issues/3496

alimirjamali committed:
Improve qvm-check mechanism on invalid domains

fixes: https://github.com/QubesOS/qubes-issues/issues/3496

alimirjamali commented:
> Attention: Patch coverage is `65.00000%` with `7 lines` in your changes missing coverage. Please review.

Ok. review bot. I will write the requested Unit Tests.

alimirjamali committed:
Improving unit tests for qvm-check

marmarek commented on qubesadmin/tools/qvm_check.py:
```diff
@@ -20,15 +20,33 @@
+
+    def error(self, message):
+        if message.startswith('no such domain: '):
+            self._invalid_domains.append(message[17:-1])
```
I don't like this approach for two reasons:
- matching error message by string is fragile
- `error` method normally terminates the program, changing this behavior is asking for troubles

Better extend QubesArgumentParser with this functionality (an option to collect invalid domains instead of exiting?), instead of catching the error after the fact.

alimirjamali replied:
> Better extend QubesArgumentParser with this functionality (an option to collect invalid domains instead of exiting?), instead of catching the error after the fact.

It could be done. But the patch will be longer. As two classes have to be extended. I will work on it.

alimirjamali replied:
It is done. I am waiting for CI/CD bot to re-evaluate it to see if any additional unit test is necessary.

alimirjamali committed:
Extended QubesArgumentParser & VmNameAction for qvm-check

According to Marek's advise:
- https://github.com/QubesOS/qubes-core-admin-client/pull/297/files#r1678613764

alimirjamali commented:
Making pylint happy(er). The original `VmNameGroup` had the same `# pylint: disable=redefined-builtin` directive.
